src/iop/cacorrect.c

```diff
@@ -1479,29 +1490,127 @@ static void CA_correct(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *pie
+      {
+        nongreen[(row / 2) * h_width + col / 2] = (in[row * width + col] <= 1.0f || oldraw[row * h_width + col / 2] <= 1.0f)
+          ? 1.0f : LIM(oldraw[row * h_width + col / 2] / in[row * width + col], 0.5f, 2.0f);
+      }
+    }
+
+    if(height % 2)
+    {
+      // odd height => factors are not set in last row => use values of preceding row
+      for(int col = 0; col < h_width; col++)
+      {
+        redfactor[(h_height-1) * h_width + col] =  redfactor[(h_height-2) * h_width + col];
+        bluefactor[(h_height-1) * h_width + col] =  bluefactor[(h_height-2) * h_width + col];
+      }
+    }
+
+    if(width % 2)
+    {
+      // odd width => factors for one channel are not set in last column => use value of preceding column
+      const int ngRow = 1 - (FC(0, 0, filters) & 1);
+      const int ngCol = FC(ngRow, 0, filters) & 1;
+      const int color = FC(ngRow, ngCol, filters);
+      float *nongreen = (color == 0) ? redfactor : bluefactor;
+      for(int row = 0; row < h_height; row++)
+      {
+        nongreen[row * h_width + h_width - 1] = nongreen[row * h_width + h_width - 2];
+      }
+    }
+
+    // blurr correction factors
+    float valmax[] = { 10.0f };
+    float valmin[] = { 0.1f };
+    dt_gaussian_t *red  = dt_gaussian_init(h_width, h_height, 1, valmax, valmin, 30.0f, 0);
+    dt_gaussian_t *blue = dt_gaussian_init(h_width, h_height, 1, valmax, valmin, 30.0f, 0);
+    if(red && blue)
+    {
+      dt_gaussian_blur(red, redfactor, redfactor);
+      dt_gaussian_blur(blue, bluefactor, bluefactor);
 
-  //   if(plistener)
-  //   {
-  //     plistener->setProgress(1.0);
-  //   }
+#ifdef _OPENMP
+  #pragma omp for
+#endif
+      for(int row = 2; row < height - 2; row++)
+      {
+        const int firstCol = FC(row, 0, filters) & 1;
+        const int color = FC(row, firstCol, filters);
+        float *nongreen = (color == 0) ? redfactor : bluefactor;
+        for(int col = firstCol; col < width - 2; col += 2)
+        {
+          const float correction = nongreen[row / 2 * h_width + col / 2];
+          out[row * width + col] *= correction;
+        }
+      }
+    }
+    if(red)  dt_gaussian_free(red);
+    if(blue) dt_gaussian_free(blue);
+  }
+
+  free(buffer1);
+  dt_free_align(RawDataTmp);
+  dt_free_align(Gtmp);
+  dt_free_align(redfactor);
+  dt_free_align(bluefactor);
+  dt_free_align(oldraw);
 }
 
 /*==================================================================================
  * end raw therapee code
  *==================================================================================*/
+static void _display_ca_error(struct dt_iop_module_t *self)
+{
+  dt_iop_cacorrect_gui_data_t *g = (dt_iop_cacorrect_gui_data_t *)self->gui_data;
+
+  if(g == NULL) return;
+   ++darktable.gui->reset;
+
+  if(g->error == CACORRECT_ERROR_CFA)
+    dt_iop_set_module_trouble_message(self, _("error"),
+                                      _("CA correction supports only RGB Colour filter arrays"), NULL);
```

Comment:
Please remove all capital letters (except for CA & RGB).

Same for all strings below.

Reply:
done directly via web